elastic/apm-server
# Logs intake fixes

## Motivation/summary

Fix some a few small bugs in the logs intake:

- accept `event.dataset`, not `dataset`
- don't set `error.grouping_key` for application logs just because they have `error.*` fields
- add missing field definitions to app_logs

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Follow steps in https://github.com/elastic/apm-server/pull/9349
2. Check that none of the logs have `error.grouping_key` set
3. Check that all fields are explicitly mapped; there should be no unknown fields when looking in Discover

## Related issues

Closes https://github.com/elastic/apm-server/issues/9666
